Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -1504,7 +1518,7 @@ bool HTMLInputElement::isRequiredFormControl() const
 
 bool HTMLInputElement::matchesReadWritePseudoClass() const
 {
-    return m_inputType->supportsReadOnly() && !isDisabledOrReadOnly();
+    return supportsReadOnly() && isMutable();
```

Comment:
Note to self: `supportsReadOnly() &&` is redundant here since `isMutable()` checks for it.